test/distributed/_sharded_tensor/ops/test_linear.py

```diff
@@ -63,8 +63,38 @@ def _run_sharded_linear(self, spec, input_size, linear_size, sharded_dim):
         sharded_output = torch.nn.functional.linear(
             inp, sharded_linear.weight, sharded_linear.bias
         )
+
         self.assertEqual(local_output, sharded_output)
 
+        # Generate expected for loss calculation.
+        expected_result = torch.randint(0, 2, (input_size[0],)).cuda(self.rank)
+        external_grad = torch.ones(input_size[0], dtype=torch.float32).cuda(self.rank).contiguous()
+
+        # Compute loss.
+        local_loss = expected_result - torch.sum(local_output, dim=1)
+        sharded_loss = expected_result - torch.sum(sharded_output, dim=1)
+        sharded_loss = sharded_loss.contiguous()
+
+        # Run backward pass
+        local_loss.backward(gradient=external_grad)
+        sharded_loss.backward(gradient=external_grad)
+
+        # Verify that both weight and bias in the sharded linear has non-None grad.
+        sharded_weight = sharded_linear.weight.local_shards()[0].tensor
+        self.assertEqual(sharded_weight.requires_grad, True)
+        self.assertNotEqual(sharded_linear.bias.grad, None)
+        self.assertNotEqual(sharded_weight.grad, None)
```

Comment:
I think one potential problem might be with the `shard_parameter` method where it doesn't preserve the `requires_grad` field. Can you check `sharded_weight.requires_grad` after `shard_parameter` to see if it is True?

Reply:
yeah this is already after `shard_parameter`.